crates/sats/src/relation.rs

```diff
@@ -147,15 +151,15 @@ pub struct ColumnOnlyField<'a> {
 // TODO(perf): Remove `Clone` derivation.
 #[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub struct Column {
-    pub field: FieldName,
+    pub field: Arc<FieldName>,
     pub algebraic_type: AlgebraicType,
     pub col_id: ColId,
 }
 
 impl Column {
     pub fn new(field: FieldName, algebraic_type: AlgebraicType, col_id: ColId) -> Self {
         Self {
-            field,
+            field: field.into(),
```

Comment:
We're passing in an owned `FieldName`, so I don't think we need to clone here.

Reply:
This is to mesh well with:

```rust
impl Fields {
    /// Creates a new instance of `Fields` with the given columns, and build an internal map of `FieldName` to their positions.
    pub fn new(columns: Vec<Column>) -> Self {
        let idx = columns
            .iter()
            .enumerate()
            .map(|(pos, x)| (x.field.clone(), pos))
            .collect();
        Self { columns, idx }
    }
```

Without the `arc` this will be another alloc.